src/Validator/Validators/hdtSCHValidator.cpp

```
@@ -109,8 +116,20 @@ namespace hdt
 				SCHViolation v;
 				v.xmlPath = xmlPath;
 				v.location = BuildNodeLocationPath(xnode.node());
-				v.line = OffsetToLineNumber(bytes, xnode.node().offset_debug());
 				v.message = resolveMessageTemplate(rule.message, xnode.node());
+
+				// The source map gives the author-meaningful line: a hand-written element's original line
+				// (expansion shifts raw line numbers when patterns are present) or, for generated content,
+				// the <pattern> use line. Only when the file used no patterns is there no map, so the
```

Cppcheck (2.21.0)

[style] The function 'updateTransformUpDown' is never used.

(unusedFunction)